### PR TITLE
Offset: report offset for zero size elements

### PR DESCRIPTION
**Pull request by poppinlp**
### Checklist ###
Mark an `[x]` for completed items, if you're not sure leave them unchecked and we can assist.

* [ ] All authors have signed the CLA at https://contribute.jquery.com/CLA/
* [ ] New tests have been added to show the fix or feature works
* [ ] Grunt build and unit tests pass locally with these changes
* [ ] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

Thanks! Bots and humans will be around shortly to check it out.

Is this fix OK?
I haven't add unit test case for offset method with zero size element since i don't know whether you need that.
For [issue 3167](https://github.com/jquery/jquery/issues/3167)

### CODE REVIEW

File: src/offset.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ jQuery.fn.extend( {
 		rect = elem.getBoundingClientRect();
 
 		// Make sure element is not hidden (display: none)
-		if ( rect.width || rect.height ) {
+		if ( elem.style.display !== 'none' ) {
```
Comment (dmethvin) [inline src/offset.js]: What if its parent is `display; none`?

We would also need a few unit tests for this case to be sure that it works now and doesn't break in the future.

File: src/offset.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ jQuery.fn.extend( {
 		rect = elem.getBoundingClientRect();
```
Comment (dmethvin) [inline src/offset.js]: For width and height measurements [we use getClientRects](https://github.com/jquery/jquery/blob/ff1a0822f72d2b39fac691dfcceab6ede5623b90/src/css.js#L124-129) to determine if the element is disconnected, before calling getBoundingClientRect, it seems like this IE11 problem might apply here too?

**Comment by mgol** [comment]
@poppinlp Do you want to finish this PR? We'd need you yo sign our [CLA](https://contribute.jquery.org/CLA/) and respond to @dmethvin's comments.

**Comment by poppinlp** [comment]
@mgol 
Sry for no response since i got something to do the last few days.
Yep, I'll finish this recently. THX.

**Comment by poppinlp** [comment]
@mgol 
Sry ag. I'll fix it this week.
I have a project must be online last week. :(

**Comment by mgol** [comment]
Sure, no problem!

-- 
Michał Gołębiowski

**Comment by timmywil** [comment]
Don't want to leave this open. It needs more work before reopening.
